lib/porky_lib/symmetric.rb

```diff
@@ -40,24 +40,37 @@ def create_alias(key_id, key_alias)
     client.create_alias(target_key_id: key_id, alias_name: key_alias)
   end
 
-  def encrypt(data, cmk_key_id, encryption_context = nil)
-    return if data.nil? || cmk_key_id.nil?
-
-    # Generate a new data encryption key
+  def generate_data_encryption_key(cmk_key_id, encryption_context = nil)
     PorkyLib::Config.logger.info('Generating new data encryption key')
-
     resp = {}
     resp = client.generate_data_key(key_id: cmk_key_id, key_spec: SYMMETRIC_KEY_SPEC, encryption_context: encryption_context) if encryption_context
     resp = client.generate_data_key(key_id: cmk_key_id, key_spec: SYMMETRIC_KEY_SPEC) unless encryption_context
 
-    plaintext_key = resp.to_h[:plaintext]
-    ciphertext_key = resp.to_h[:ciphertext_blob]
+    [resp.to_h[:plaintext], resp.to_h[:ciphertext_blob]]
+  end
+
+  def decrypt_data_encryption_key(ciphertext_key, encryption_context = nil)
+    PorkyLib::Config.logger.info('Decrypting data encryption key')
```

Comment:
Reduce to:

`return client.decrypt(ciphertext_blob: ciphertext_key, encryption_context: encryption_context).to_h[:plaintext] if encryption_context


client.decrypt(ciphertext_blob: ciphertext_key).to_h[:plaintext]`